File: python/leetcode/0013-roman-to-integer/solution01.py
ROMAN = {
    'I': 1,
    'V': 5,
    'X': 10,
    'L': 50,
    'C': 100,
    'D': 500,
    'M': 1000,
}

class Solution:
    def romanToInt(self, s: str) -> int:
        num = 0
        index = 0

        while index < len(s):
            current_str = s[index]
            next_str = ""

            if index + 1 < len(s):
                next_str = s[index + 1]

            current_value = ROMAN.get(current_str, 0)
            next_value = ROMAN.get(next_str, 0)

            if current_value < next_value:
                index += 1
                num += next_value - current_value
            else:
                num += current_value

            index += 1

        return num

File: typescript/leetcode/0013-roman-to-integer/solution01.ts
const ROMAN: {[key: string]: number} = {
    'I': 1,
    'V': 5,
    'X': 10,
    'L': 50,
    'C': 100,
    'D': 500,
    'M': 1000,
}

/**
 * @param {string} s
 * @return {number}
 */
export function romanToInt(s: string): number {
  let num = 0
  let index = 0

  while (index < s.length) {
    const currentValue = ROMAN[s[index]]
    const nextValue = ROMAN[s[index + 1]] || 0

    if (currentValue < nextValue) {
      num += nextValue - currentValue
      index++
    } else {
      num += currentValue
    }

    index++
  }

  return num
}
